src/sources/exec/mod.rs

```diff
@@ -419,32 +410,45 @@ async fn run_command(
         .ok_or_else(|| Error::new(ErrorKind::Other, "Unable to take stdout of spawned process"))?;
 
     // Create stdout async reader
-    let stdout = stdout.allow_read_until(shutdown.clone().map(|_| ()));
     let stdout_reader = BufReader::new(stdout);
 
     let pid = child.id();
 
     spawn_reader_thread(stdout_reader, decoder.clone(), STDOUT, sender);
 
-    while let Some(((mut events, byte_size), stream)) = receiver.recv().await {
-        emit!(BytesReceived {
-            byte_size,
-            protocol: "exec",
-        });
-
-        let count = events.len();
-        emit!(ExecEventsReceived {
-            count,
-            command: config.command_line().as_str(),
-            byte_size: events.size_of(),
-        });
-
-        for event in &mut events {
-            handle_event(&config, &hostname, &Some(stream.to_string()), pid, event);
-        }
-        if let Err(error) = out.send_batch(events).await {
-            emit!(StreamClosedError { count, error });
-            break;
+    'outer: loop {
```

Comment:
I don't think I've seen this exact syntax, doesn't look like a lifetime - what is it?

Reply:
Ah, it's a loop label that you can use to break out of nested loops using: https://doc.rust-lang.org/rust-by-example/flow_control/loop/nested.html

Reply:
TIL - thanks!